tests/integration/helpers.py

```diff
@@ -139,3 +151,32 @@ async def http_request(
             return resp.status_code
 
         return resp.json()
+
+
+@retry(
+    wait=wait_fixed(wait=5) + wait_random(0, 5),
+    stop=stop_after_attempt(15),
+)
+async def check_cluster_formation_successful(
+    ops_test: OpsTest, unit_ip: str, unit_names: List[str]
+) -> bool:
+    """Returns whether the cluster formation was successful and all nodes successfully joined.
+
+    Args:
+        ops_test: The ops test framework instance.
+        unit_ip: The ip of the unit of the OpenSearch unit.
+        unit_names: The list of unit names in the cluster.
+
+    Returns:
+        Whether The cluster formation is successful.
+    """
+    response = await http_request(ops_test, f"https://{unit_ip}:9200/_nodes", "GET")
+    if "_nodes" not in response or "nodes" not in response:
```

Comment:
Why the double-check for `'nodes'` substring?

Reply:
Ah, type-hint. Could avoid with a `.get('nodes', {})` on L181, but OK if not.

Reply:
The first key starts with a `_` and the second without, the response of that HTTP call essentially looks like:
```
{
    "_nodes": ...
    "nodes": ....
}
```